Willmish/schedule multi gpu

Willmish opened the pull request:
- [ ] Script tuned for scheduling unlearing of specified task, according to the plan: https://github.com/Adamliu1/SNLP_GCW/issues/98 (128, 512, 1024 sequential for splits 4, 16, 64; 128, 512, 1024 batch; 20480 samples continuous)
- [ ] Tuned to be ran on a 4 GPU node (could be expanded/adapted to fewer, but requires manual tweaking)

Willmish committed:
Add scheduling script for 4 gpu unlearning.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Working for opt1.3b on dip machine (A6000)

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Fix checkpointing for batch (every epoch, already aligned to cont.)

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add date calls to the script.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add lr finding scripcior.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Project: Hail Mary.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Fix lm_eval harness scheduling. Add continous unlearning eval on separate node,:

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish commented:
follows from #109

Adamliu1 committed:
integrated accelerate and deepspeed, logging fixing WIP

Willmish committed:
Deepspeed with llama working ?

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Adjust model precision by default to bf16.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Update lr finding for continuous unlearning script to use deepspeed.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Davidyz commented on llm_unlearn_ucl/unlearn_harm.py:
```diff
@@ -477,26 +516,34 @@ def main(args) -> None:
-    )
-    wandb.log_artifact(data_sample_artifacts)
+    # TODO: TEST THIS
+    if bool(args.wandb_log) and accelerator.is_local_main_process:
```
This should prob go [here](https://github.com/Adamliu1/SNLP_GCW/blob/380f2e755a7893f77a80c8adab6d4f461d2a2c1e/llm_unlearn_ucl/data_utils.py#L66)

Davidyz reviewed (changes requested):
The code related to uploading dataset samples to wandb can go to [`data_utils.py`](https://github.com/Adamliu1/SNLP_GCW/blob/380f2e755a7893f77a80c8adab6d4f461d2a2c1e/llm_unlearn_ucl/data_utils.py#L66) because the files containing the samples are created there.

Willmish committed:
Cleanup unlearn_harm.py.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Cleanup.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add schedule_unlearning_script for 1e-7.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Update script for 1e-7.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add schedule_unlearning_script for 5e-7 (sequential).

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Update Batch size to batch size 2..

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add schedule_unlearning_script for 1e-6 (sequential).

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Fix issues with scheduling scripts + SKIP unlearning 128 Sample count (happend on first attempt with faulty script).

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add special case for saving model when running with ZERO-3 optimisation.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
:art: Format Python code with psf/black

Willmish committed:
Multi-gpu sequentual longer runs.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add deepspeed 4 gpu config.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Update unlearn script to schedule multi-epoch experiment. Change folder structure.:

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add script for multi-epcoh seq expriment for 1e-5 LR.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Change eval_dramework_tasks eval_results.py save path.

Fix for single result file processing, by NOT extracting the idx from the filename.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
:art: Format Python code with psf/black

Willmish committed:
Add script for scheduling multi epoch seq unlearning for lr 5e-6.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

TheRootOf3 committed:
Added max sample length functionality.

Signed-off-by: TheRootOf3 <aceszablewski@gmail.com>

Willmish committed:
Add french unlearning sched scripts.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Remove code for printing sample length stats in unlearn dataset.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>

Willmish committed:
Add parameters for loading normal dataset with updated DataloaderConstructor.

Signed-off-by: Szymon Duchniewicz <szymon.duchniewicz.20@ucl.ac.uk>
